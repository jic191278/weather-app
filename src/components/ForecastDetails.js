import React from 'react';

import '../styles/forecast-details.css';
// import east from '../images/E.png';

const ForecastDetails = ({ forecast }) => {
  const { date, humidity, temperature, wind } = forecast;
  const { max, min } = temperature;
  const { direction, speed } = wind;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <ul className="forecast-details__detail-items">
        <li>
          <h3 className="forecast-details__date">{formattedDate}</h3>
        </li>
        <li className="forecast-details__detail-item">
          <span className="forecast-details__detail-item--label">
            Max Temperature:
          </span>
          <span className="forecast-details__detail-item--value">{`${max}°c`}</span>
        </li>
        <li className="forecast-details__detail-item">
          <span className="forecast-details__detail-item--label">
            Min Temperature:
          </span>
          <span className="forecast-details__detail-item--value">{`${min}°c`}</span>
        </li>
        <li className="forecast-details__detail-item">
          <span className="forecast-details__detail-item--label">
            Humidity:
          </span>
          <span className="forecast-details__detail-item--value">
            {`${humidity}%`}
          </span>
        </li>

        <li className="forecast-details__detail-item">
          <div className="forecast-details__detail--wind-container">
            <span className="forecast-details__detail-item--label">Wind:</span>
            <span className="forecast-details__detail-item--value">{`${speed}mph`}</span>
            <span>
              <img
                className="forecast-details__detail-item--wind-direction"
                data-testid="wind-direction"
                src={`images/${direction}.png`}
                alt="wind direction icon"
              />
            </span>
          </div>
        </li>
      </ul>
    </div>
  );
};

export default ForecastDetails;

// Needs
// - date
// - temp - max & min
// - humidity
// - wind speed & direction
