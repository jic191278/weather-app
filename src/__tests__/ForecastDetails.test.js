import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ForecastDetails from '../components/ForecastDetails';

describe('ForecastDetails', () => {
  beforeEach(() => {
    cleanup();
  });

  const forecastProp = {
    date: 1710530097000, // Fri Mar 15 2024
    humidity: 10,
    temperature: {
      max: 15,
      min: 12,
    },
    wind: {
      direction: 'w',
      speed: 25,
    },
  };

  describe('ForecastDetails Snapshot', () => {
    const { asFragment } = render(<ForecastDetails forecast={forecastProp} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Component elements render', () => {
    beforeEach(() => {
      render(<ForecastDetails forecast={forecastProp} />);
    });

    test('Correct date renders', () => {
      const dateListItem = screen.getByText('Fri Mar 15 2024');

      expect(dateListItem).toBeInTheDocument();
    });

    test('Max temperature renders', () => {
      const maxTempListItem = screen.getByText(/15°c/i);

      expect(maxTempListItem).toBeInTheDocument();
    });

    test('Min temperature renders', () => {
      const minTempListItem = screen.getByText(/12°c/i);

      expect(minTempListItem).toBeInTheDocument();
    });

    test('Humidity renders', () => {
      const humidityTempListItem = screen.getByText(/10%/i);

      expect(humidityTempListItem).toBeInTheDocument();
    });

    test('Wind speed renders', () => {
      const windSpeedTempListItem = screen.getByText(/25mph/i);

      expect(windSpeedTempListItem).toBeInTheDocument();
    });

    test('Wind direction renders', () => {
      const windDirectionListItem = screen.getByTestId('wind-direction');

      expect(windDirectionListItem).toBeInTheDocument();
    });
  });
});
