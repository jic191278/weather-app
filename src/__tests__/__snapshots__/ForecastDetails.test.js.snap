// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <ul
      class="forecast-details__detail-items"
    >
      <li>
        <h3
          class="forecast-details__date"
        >
          Fri Mar 15 2024
        </h3>
      </li>
      <li
        class="forecast-details__detail-item"
      >
        <span
          class="forecast-details__detail-item--label"
        >
          Max Temperature:
        </span>
        <span
          class="forecast-details__detail-item--value"
        >
          15°c
        </span>
      </li>
      <li
        class="forecast-details__detail-item"
      >
        <span
          class="forecast-details__detail-item--label"
        >
          Min Temperature:
        </span>
        <span
          class="forecast-details__detail-item--value"
        >
          12°c
        </span>
      </li>
      <li
        class="forecast-details__detail-item"
      >
        <span
          class="forecast-details__detail-item--label"
        >
          Humidity:
        </span>
        <span
          class="forecast-details__detail-item--value"
        >
          10%
        </span>
      </li>
      <li
        class="forecast-details__detail-item"
      >
        <div
          class="forecast-details__detail--wind-container"
        >
          <span
            class="forecast-details__detail-item--label"
          >
            Wind:
          </span>
          <span
            class="forecast-details__detail-item--value"
          >
            25mph
          </span>
          <span>
            <img
              alt="wind direction icon"
              class="forecast-details__detail-item--wind-direction"
              data-testid="wind-direction"
              src="images/w.png"
            />
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
